src/Dialect/ONNX/ONNXDimAnalysis.cpp

```diff
@@ -941,82 +941,79 @@ void DimAnalysis::visitDim(
     Value data = reshapeOp.getData();
     Value output = reshapeOp.getReshaped();
 
-    // Special case 1: the output dimension i can be from
-    // - data[j] if
-    //    - dim j and i are the only dynamic dimension in data and output,
-    //    respectively, and
+    // Special case: when data and output have the same number of dynamic
+    // dimensions (N), the dynamic dimension output[i] is the same as the
+    // dynamic dimension data[j] if
     //    - the products of the remaining static dimensions in data and output
-    //    are equal.
+    //    are equal, and
+    //    - the remaining N-1 dynamic dimensions in data and output are the
+    //    same, respectively.
```

Comment:
Just to understand, isn't that the same as saying that all but 1 dynamic dimensions can be shown to be the same (where the other dims are either static or dynamic)? That feel clearer to me.

Reply:
> isn't that the same as saying that all but 1 dynamic dimensions can be shown to be the same (where the other dims are either static or dynamic)?

Yes, they are the same but for dynamic only. For static dims, we compare the accumulation of all static dims.

> So we can accumulate all static sizes, and pairwise match dynamic regardless of position in the rank vector, and if the last 2 are dynamic, then they must be equal. Is that what you are doing? Above, d1 and d2 match, then d0 == d3.

Yes, it is what I am doing. The position of dynamic dims does not matter.